unit/analyses/call_graph.cpp

```diff
@@ -0,0 +1,109 @@
+
+#include <iostream>
+
+#include <catch.hpp>
+
+#include <analyses/call_graph.h>
+
+#include <util/symbol_table.h>
+#include <util/std_code.h>
+
+#include <goto-programs/goto_convert_functions.h>
+
+static symbolt create_void_function_symbol(
+  const irep_idt &name,
+  const codet &code)
+{
+  code_typet void_function_type;
+  symbolt function;
+  function.name=name;
+  function.type=void_function_type;
+  function.mode=ID_java;
+  function.value=code;
+  return function;
+}
+
+static bool multimap_key_matches(
+  const std::multimap<irep_idt, irep_idt> &map,
+  const irep_idt &key,
+  const std::set<irep_idt> &values)
+{
+  auto matching_values=map.equal_range(key);
+  std::set<irep_idt> matching_set;
+  for(auto it=matching_values.first; it!=matching_values.second; ++it)
+    matching_set.insert(it->second);
+  return matching_set==values;
+}
+
+SCENARIO("call_graph",
+  "[core][util][call_graph]")
+{
+
+  GIVEN("Some cyclic function calls")
+  {
+
+    symbol_tablet symbol_table;
```

Comment:
If you wouldn't mind adding just a comment showing the code generated I think it would make the whole test a whole lot clearer.

```C++
// void A() 
// { 
//    A(); 
//    B(); 
// }
// void B() 
// { 
//   C(); 
//   D(); 
// }
// void C() { }
// void D() { }
```